terasoluna-gfw-functionaltest-selenium/src/test/java/org/terasoluna/gfw/functionaltest/app/redirect/RedirectTest.java

```diff
@@ -108,20 +109,47 @@ public void test01_03_redirectToExternalLink() {
 
         driver.findElement(By.id("btn1")).click();
 
-        // Unlike other application servers, WebSphere Liberty Profile & WebSphere traditional wraps an unexpected exception of
-        // ServletException.
-        // So the expected error page is different.
-        String expectedErrorMessage;
-        if (apServerName == ApServerName.WEBSPHERELP
-                || apServerName == ApServerName.WEBSPHERETR) {
-            expectedErrorMessage = "System Error...";
+        // Unlike other application servers, in Tomcat, the Location of the response header is empty when sendRedirect ("").
+        // So the expected redirected URL is different.
+        // Details, see https://github.com/terasolunaorg/terasoluna-gfw-functionaltest/issues/855
+        String expectedURL;
+        if (apServerName == ApServerName.TOMCAT) {
+            expectedURL = applicationContextUrl + "/login";
         } else {
-            expectedErrorMessage = "Page Not Found";
+            expectedURL = applicationContextUrl + "/";
         }
 
-        // confirms that 404 error occurred after login transition
-        assertThat(driver.findElement(By.xpath("/html/body/div/h2")).getText(),
-                is(expectedErrorMessage));
+        urlToBe(expectedURL);
+
+        // check URL.
+        assertThat(driver.getCurrentUrl(), is(expectedURL));
+
+        driver.get(getPackageRootUrl());
```

Comment:
- [ ] It seems good to add comments to explain following test code set.

Reply:
35c9eef561ccab79387798f39e204a3686a335f0

I add comment above commit .

Reply:
You should add explanations not only in the second half but also in the first half.

Reply:
7d8d366cbfab2cd951024168576e539ab2d0fed1

I also added a comment at the beginning of the test case.